### PR TITLE
adapting the sdk liquidations to the new FSL contract interface

### PR DESCRIPTION
**Pull request by vminkov**
## Description

adapting the sdk liquidations to the new FSL contract interface

### CODE REVIEW

File: packages/sdk/chainDeploy/helpers/liquidators/fuseSafeLiquidator.ts [hunk 1]
```diff
@@ -76,6 +76,17 @@ export const configureFuseSafeLiquidator = async ({
     }
   }
 
+  for (const address in chainIdToConfig[chainId].fundingStrategies) {
+    const [redemptionStrategyType] = chainIdToConfig[chainId].fundingStrategies[address];
```
Comment (carlomazzaferro) [inline packages/sdk/chainDeploy/helpers/liquidators/fuseSafeLiquidator.ts:80]: `fundingStrategies[address]` returns `[FundingStrategyContract, string]` right? I hope the destructuring of `const [redemptionStrategyType] =` works fine, just wondering if it doesn't need smt like `const [redemptionStrategyType, ] =`

File: packages/sdk/chainDeploy/mainnets/polygon.ts [hunk 2]
```diff
@@ -558,9 +558,13 @@ export const deploy = async ({ run, ethers, getNamedAccounts, deployments }: Cha
 
   /// Addresses Provider - set bUSD
   const addressesProvider = (await ethers.getContract("AddressesProvider", deployer)) as AddressesProvider;
-  const tx = await addressesProvider.setAddress("bUSD", assets.find((a) => a.symbol === assetSymbols.BUSD)!.underlying);
-  await tx.wait();
-  console.log("setAddress: ", tx.hash);
+  const busdAddress = assets.find((a) => a.symbol === assetSymbols.BUSD)!.underlying;
```
Comment (carlomazzaferro) [inline packages/sdk/chainDeploy/mainnets/polygon.ts:561]: why do we store bUSD on polygon?

File: packages/sdk/deploy/deploy.ts [hunk 3]
```diff
@@ -460,26 +460,55 @@ const func: DeployFunction = async ({ run, ethers, getNamedAccounts, deployments
...
+  if (fpdAddress !== fpd.address) {
+    tx = await addressesProvider.setAddress("FusePoolDirectory", fpd.address);
+    await tx.wait();
+    console.log("setAddress FusePoolDirectory: ", tx.hash);
+  }
 
-  tx = await addressesProvider.setAddress("FuseFeeDistributor", ffd.address);
-  await tx.wait();
-  console.log("setAddress: ", tx.hash);
+  const ffdAddress = await addressesProvider.callStatic.getAddress("FuseFeeDistributor");
```
Comment (carlomazzaferro) [inline packages/sdk/deploy/deploy.ts:506]: should we also set the FSL in the ap?

Comment (vminkov) [reply packages/sdk/deploy/deploy.ts:506]: yeah, good idea

File: packages/sdk/chainDeploy/mainnets/polygon.ts (hunk 2 above)
Comment (vminkov) [reply packages/sdk/chainDeploy/mainnets/polygon.ts:561]: I have idea, I thought you added it

**Review by carlomazzaferro — COMMENTED** [review]
just a bunch of questions, but LG

File: packages/sdk/chainDeploy/mainnets/polygon.ts (hunk 2 above)
Comment (carlomazzaferro) [reply packages/sdk/chainDeploy/mainnets/polygon.ts:561]: let's keep it as I think we use it in the tests

**Pull request merged**